crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -82,13 +82,27 @@ impl FileAssetIo {
     pub fn root_path(&self) -> &PathBuf {
         &self.root_path
     }
+
+    #[cfg(not(windows))]
+    fn ensure_scope(&self, target: &Path) -> Result<PathBuf, AssetIoError> {
```

Comment:
hmm not super keen on adding this to our general "path validation / checking" logic (which is peppered throughout high-traffic asset loading apis). Canonicalize will have some overhead (and allocates a new PathBuf), then "starts_with" does even more "full string value comparisons".

When weighing that consistent runtime cost against symlinking weirdness, I think I'd prefer to just let symlinks be weird and strongly recommend against them in our docs.